### DISCUSSION TITLE
How to host

### DISCUSSION THREAD

**Discussion by teosssss** [go-telegram-bot-api/telegram-bot-api#502] [Q&A]
How can i host my bot, can I host it with heroku? or there are other methods for free?

**Comment by watn3y** [comment]
Free is always tricky, but probably doable. How? No idea.
I would recommend you get yourself a cheap VPS somewhere and host your bot there, cheap is always miles better than free in my opinion.
Regarding Heroku, the [official docs ](https://devcenter.heroku.com/articles/getting-started-with-go)seem like the best place to start.

EDIT:
You can always run your bot on something like a Raspberry Pi Zero.
